SpatialGDK/Source/SpatialGDKServices/Private/LocalDeploymentManager.cpp

```diff
@@ -43,11 +43,20 @@ FLocalDeploymentManager::FLocalDeploymentManager()
 		// Watch the worker config directory for changes.
 		StartUpWorkerConfigDirectoryWatcher();
 
-		// Restart the spatial service so it is guaranteed to be running in the current project.
 		AsyncTask(ENamedThreads::AnyBackgroundThreadNormalTask, [this]
 		{
+			// Stop existing spatial service to guarantee that any new existing spatial service would be running in the current project.
 			TryStopSpatialService();
-			TryStartSpatialService();
+
+			// Start spatial service in the current project if spatial networking is enabled
+			if (GetDefault<UGeneralProjectSettings>()->bSpatialNetworking)
+			{
+				TryStartSpatialService();
+			}
+			else
+			{
+				UE_LOG(LogSpatialDeploymentManager, Verbose, TEXT("SpatialOS deployment not started because spatial networking is disabled."));
+			}
 
 			// Ensure we have an up to date state of the spatial service and local deployment.
 			RefreshServiceStatus();
```

Comment:
Shouldn't we also only do this if running with Spatial networking? Otherwise I assume this will be spamming the output log with errors that spatiald isn't running.

Reply:
I can imagine that it would be appropriate for the status to always be logged regularly if the user has chosen to enable the SpatialOS plugin.  Knowing whether it's on or off might help with debugging. It's also set to Verbose, where a user might expect to get every bit of information that can be helpful.

Otherwise, we would be turning this logging service (/debugging helper) on and off based on when we personally *expect* for the spatial daemon to be running.